### PR TITLE
INTYGFV-15592: Added total number of sick leaves to populated filters…

### PR DESCRIPTION
**Pull request by mhornfeldt**
… response and refactored filtering on protected person

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/PuFilterServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+            final var personSvar = personSvarMap.get(pnr.get());
+            final var patientNotFound = personSvar.getStatus() == PersonSvar.Status.NOT_FOUND;
+            if (personSvar.getStatus() == PersonSvar.Status.FOUND || patientNotFound) {
+                if (patientNotFound || personSvar.getPerson().isSekretessmarkering()) {
+
+                    final var updatedName = patientNotFound ? SEKRETESS_SKYDDAD_NAME_UNKNOWN : SEKRETESS_SKYDDAD_NAME_PLACEHOLDER;
+                    item.setPatientNamn(updatedName);
+
+                    if (filterProtectedPerson) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/PuFilterServiceImpl.java:54]: The tests you found in Rehabstod needs to be covered here:
- testSekretessmarkeradIsIncludedWhenUserIsLakareAsRehabkoordinatorAndSignedTheIntyg
- testSekretessmarkeradIsExcludedWhenUserIsLakareAsRehabkoordinatorOnSameUnitButIsNotSigning

The rule is that only a doctor should see any protected patient if the doctor was the issuer of the sick leave certificate.

What do you think about a solution like this?
filterProtectedPerson change from boolean -> string. It contains the doctorId that we should allow to show protected patients for. If no doctorId is provided, no protected patients should be displayed/return. If a doctorId is provided it needs to match the doctorId on the IntygData.

filterProtectedPerson could change to something that describes the use a bit better. Don't have a great naming suggestion yet :-)

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nicely done! A few comments to look at before it is ready for merge.

**Comment by mhornfeldt** [comment]
> Nicely done! A few comments to look at before it is ready for merge.

Fixed!

**Review by mww-sopra-steria — APPROVED** [review]
Nice, approved!

**Pull request merged**